source/fundamentals/gridfs.txt

```diff
@@ -0,0 +1,278 @@
+below uses the ``forEach()`` method to display the results:
+
+.. code-block:: javascript
+
+   const cursor = bucket.find({});
+   cursor.forEach(doc => console.log(doc));
+
+The ``find()`` method accepts various query specifications and can be
+combined with other methods such as ``sort()``, ``limit()``, and ``project()``.
+
+For more information on the classes and methods mentioned in this section,
+see the following resources:
+
+- :node-api-4.0:`find() API documentation <classes/gridfsbucket.html#find>`
+- :node-api-4.0:`FindCursor API documentation <classes/findcursor.html>`
+- :doc:`Cursor Fundamentals page </fundamentals/crud/read-operations/cursor>`
+- :doc:`Read Operations page </fundamentals/crud/read-operations/>`
+
+.. _gridfs-download-files:
+
+Download Files
+--------------
+
+You can download files from your MongoDB database using a download
+stream. ``GridFSBucket`` contains the ``openDownloadStreamByName()``
+method which creates a download stream. 
+
+The example below shows you how to download a file referenced
+by the file name, stored in the ``filename`` field, into your working
+directory. Alternatively, you can use the ``openDownloadStream()``
+method, which takes the ``_id`` field of a file as a parameter. Note
+that if there are multiple documents with the same ``filename`` value,
+GridFS will stream the most recent file with the given name (as
+determined by the ``uploadDate`` field).
+
+.. code-block:: javascript
+
+   bucket.openDownloadStreamByName('myFile').
+        pipe(fs.createWriteStream('./outputFile'));
+
+.. note::
+
+   The GridFS streaming API cannot load partial chunks. When a download
+   stream needs to pull a chunk from MongoDB, it pulls the entire chunk
+   into memory. The 255 kilobyte default chunk size is usually
+   sufficient, but you can reduce the chunk size to reduce memory
+   overhead.
+
+For more information on the ``openDownloadStreamByName()`` method, see
+its :node-api-4.0:`API documentation <classes/gridfsbucket.html#opendownloadstreambyname>`.
+
+.. _gridfs-rename-files:
+
+Rename Files
+------------
+
+You can update the name of a GridFS file in your bucket by calling the
+``rename()`` method. You must specify the file to rename by its ``_id`` field
+rather than its file name.
+
+.. note::
+
+   The ``rename()`` method only supports updating the name of one file at
+   a time. To rename multiple files, retrieve a list of files matching the
+   file name from the bucket, extract the file id values from the files you
+   want to rename, and pass each file id in separate calls to the ``rename()``
+   method.
+
+The code example will show you how to update the name of the file
+referenced in this document:
+
+.. code-block:: javascript
+
+   {
+     _id: new ObjectId("60edece5e06275bf0463aaf3"),
+     length: 31851832,
+     chunkSize: 261120,
+     uploadDate: 2021-07-13T19:43:34.237Z,
+     filename: 'myFile',
+   }
```

Comment:
I don't think this adds any value to the reader (and should be removed) since you already explain you need the id of a document and your example below shows that.

Reply:
accepted. do you think it's clear enough that a person needs to replace the ObjectId("...") with their own documents id?

Reply:
It should be -- throughout our docs, we've used `ObjectId` without providing a sample document. Essentially, the user should know it's the default `_id` generated by the driver.